x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all/index.tsx

```diff
@@ -118,11 +119,15 @@ export const AllRules = React.memo<AllRulesProps>(
     const history = useHistory();
     const [, dispatchToaster] = useStateToaster();
 
-    const setRules = useCallback((newRules: Rule[]) => {
+    const setRules = useCallback((newRules: Rule[], newPagination: Partial<PaginationOptions>) => {
       dispatch({
         type: 'setRules',
         rules: newRules,
       });
+      dispatch({
```

Comment:
I am wondering if it will be better to add the pagination attributes in the action `setRules` since we need to update the pagination each time we set the rules so we can avoid two updates in our store ( meaning we will re-render twice).

Reply:
I thought about that. I did that in my second commit and then switched it to this in my latest commit. I think you're right, I should put it under `setRules` since they'll go hand in hand. I can't think of a time where we would update rules and not pagination.